tests/models/test_responses.py

```diff
@@ -190,7 +190,7 @@ def test_response_force_encoding():
 
 @pytest.mark.asyncio
 async def test_streaming_response():
-    response = httpx.Response(200, content=async_streaming_body())
+    response = httpx.Response(200, stream=async_streaming_body())
```

Comment:
This *works*, but it isn't actually valid from a type hint point of view, correct? Async generators are missing the `.aread()` method.

(Albeit tedious, I tend to be in favor of type-hinting/type-checking tests. Helps catch this kind of minor inconsistency as well as see how users will actually interact with our code *including type hints*.)

Reply:
> I tend to be in favor of type-hinting/type-checking tests.

Yeah, I've come to the view that it'd *probably* be a good plan for us, since there's *loads* of stuff that it would help catch.

(I'd need to see how things look in order to take a more informed view on it, but it *feels* like it'd be a good thing to be doing.)